src/aiq/authentication/oauth2/oauth2_auth_code_flow_provider_config.py

```diff
@@ -13,10 +13,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
+from urllib.parse import urlparse
+
 from pydantic import Field
+from pydantic import ValidationInfo
+from pydantic import field_validator
 
+from aiq.authentication.exceptions.auth_code_grant_exceptions import AuthCodeGrantConfigAuthorizationUrlFieldError
+from aiq.authentication.exceptions.auth_code_grant_exceptions import AuthCodeGrantConfigClientIDFieldError
+from aiq.authentication.exceptions.auth_code_grant_exceptions import AuthCodeGrantConfigClientSecretFieldError
+from aiq.authentication.exceptions.auth_code_grant_exceptions import AuthCodeGrantConfigClientServerUrlFieldError
+from aiq.authentication.exceptions.auth_code_grant_exceptions import AuthCodeGrantConfigScopeFieldError
```

Comment:
Lets re-use exceptions. These could all be handed with standard exceptions. Creating a new exception for each field makes it hard to catch these specific exceptions and is a lot of extra work.